testsuite/forge-cli/src/main.rs

```diff
@@ -534,43 +521,59 @@ fn single_test_suite(test_name: &str) -> Result<ForgeConfig<'static>> {
                         TransactionType::NftMint
                     }),
             )
-            .with_success_criteria(SuccessCriteria::new(
-                4000,
-                10000,
-                true,
-                Some(Duration::from_secs(240)),
-                None,
-                None,
-            )),
+            .with_success_criteria(
+                SuccessCriteria::new(4000)
+                    .add_no_restarts()
+                    .add_wait_for_catchup_s(240)
+                    .add_chain_progress(StateProgressThreshold {
+                        max_no_progress_secs: 20.0,
+                        max_round_gap: 6,
+                    }),
+            ),
         // TODO: Add tracing latency of high-gas-fee transactions
         "graceful_overload" => config
             .with_initial_validator_count(NonZeroUsize::new(10).unwrap())
-            // if we have smaller number of full nodes, TPS drops.
+            // if we have full nodes for subset of validators, TPS drops.
             // Validators without VFN are proposing almost empty blocks,
             // as no useful transaction reach their mempool.
             // something to potentially improve upon.
-            .with_initial_fullnode_count(8)
+            // So having VFNs for all validators
+            .with_initial_fullnode_count(10)
             .with_network_tests(vec![&PerformanceBenchmarkWithFN])
-            .with_emit_job(EmitJobRequest::default().mode(EmitJobMode::ConstTps { tps: 15000 }))
+            .with_emit_job(
+                EmitJobRequest::default()
+                    .mode(EmitJobMode::ConstTps { tps: 15000 })
+                    .transaction_mix(vec![
+                        (TransactionType::Transfer, 98),
```

Comment:
I am okay with the current approach of adding per type stats to the emitter but may be a simpler approach could have been just spawning two txn emittters with different gas price in the test itself?